From 6b99529b6ea66c1e318e3e95f9634841152f8849 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Tue, 14 Sep 2021 07:55:01 +0300
Subject: [PATCH] [SW-54874] gaudi2: arc: add sram offsets for different target

Since SRAM needs to be used to load preboot, this patch add SRAM
offsets for different targets (ONIX and PLDM/ASIC).

Change-Id: Idf0a3b27b49203915e4b9bc2813e03bd816a2c53
---

diff --git a/arch/gaudi2/arc/common.h b/arch/gaudi2/arc/common.h
index d7a4471..6635479 100644
--- a/arch/gaudi2/arc/common.h
+++ b/arch/gaudi2/arc/common.h
@@ -17,13 +17,21 @@
 #define HL_AXI_GENERAL_BASE	0x1000007FF0000000ULL
 #define HL_SP_SRAM_BASE		0x1000007FF7FE0000ULL
 #define HL_SP_SRAM_SIZE		0x10000
+
+#ifdef TARGET_IS_ONIX
+#define SDRAM_BASE              0x10000000
+#define SDRAM_FULL_BASE         0x1001000200000000ull
+#else
 /*
  * The SRAM region in ARC address extension is mapped to 0x10000000, but the
  * first 16MB of that region are not populated with SRAM H/W, so the actual
  * SRAM array starts at offset of 16MB from the region start and is of 48MB size
  * That makes the SRAM reside in range of 0x11000000 - 0x13FFFFFF (inclusive).
  */
-#define SDRAM_BASE		0x11000000
+#define SDRAM_BASE              0x11000000
+#define SDRAM_FULL_BASE         0x1000FFFFFD000000ull
+#endif
+
 #define GEN_REGION_BASE		0x20000000
 #define PCIE_FW_SRAM_BASE	(GEN_REGION_BASE + 0x7FF0000)
 #define FW_SRAM_BASE		(GEN_REGION_BASE + 0x7FD0000)
